### PR TITLE
net-analyzer/mk-livestatus: Move to EAPI 6

### PR DESCRIPTION
**Pull request by bjacquin**
Package-Manager: portage-2.2.28

### CODE REVIEW

File: net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,130 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
```
Comment (doughdemon) [inline net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild]: something seems off here...

File: net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+GENTOO_DEPEND_ON_PERL=no
+PERL_EXPORT_PHASE_FUNCTIONS=no
+PYTHON_COMPAT=( python2_7 )
+
+inherit autotools perl-module python-r1 eutils
```
Comment (doughdemon) [inline net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild]: eutils seems unecessary

File: net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	"${FILESDIR}/${P}-no-strip.diff"
+	"${FILESDIR}/${P}-test-RequireRcsKeywords.diff"
+)
+
+src_prepare() {
+	default
+
+	# Use system Module::Install instead, it will be copied to $S by
+	# Module::install itself.
+	rm -rf api/perl/inc
```
Comment (doughdemon) [inline net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild]: || die

File: net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+	if use perl; then
+		perl-module_src_prepare
```
Comment (doughdemon) [inline net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild]: this also calls ``default``. won't the patches now be applied twice?

File: net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+	fi
+
+	# Script too old
+	rm -f missing
```
Comment (doughdemon) [inline net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild]: || die

File: net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+	eautoreconf
+}
+
+src_configure() {
+	econf
+
+	if use perl; then
+		cd api/perl/
```
Comment (doughdemon) [inline net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild]: || die

File: net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+		perl-module_src_configure
+	fi
+}
+
+src_compile() {
+	emake
+
+	if use perl; then
+		cd api/perl
```
Comment (doughdemon) [inline net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild]: || die

File: net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+		perl-module_src_compile
+	fi
+}
+
+src_test() {
+	if use perl; then
+		cd api/perl
```
Comment (doughdemon) [inline net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild]: || die

File: net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+		export TEST_AUTHOR="Test Author"
+		perl-module_src_test
+	fi
+}
+
+src_install() {
+	emake -C src/ DESTDIR="${ED}" install-binPROGRAMS install-data-local
+
+	if use perl; then
+		cd api/perl
```
Comment (doughdemon) [inline net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild]: || die. You might want to use ``pushd`` and ``popd`` instead, though.

File: net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild [hunk 10]
```diff
@@ -0,0 +1,130 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
+
+GENTOO_DEPEND_ON_PERL=no
+PERL_EXPORT_PHASE_FUNCTIONS=no
+PYTHON_COMPAT=( python2_7 )
```
Comment (doughdemon) [inline net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild]: Could you test compatibility with Python 3? With more then one python implementation the use of the python-r1 eclass would make more sense.

**Comment by doughdemon** [comment]
Since the old ebuild was not stable on any arch, I think you can remove it. Then github could show a nicer diff.

**Comment by doughdemon** [comment]
Also, the package is maintainer needed. Would you be willing to proxy-maintain. If so, update ``metadata.xml`` according to https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide#How_proxy_maintainership_works and notify proxy-maint.

File: net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild (hunk 10 above)
Comment (bjacquin) [inline net-analyzer/mk-livestatus/mk-livestatus-1.2.2_p3-r1.ebuild]: I did some tests with older and newer version, only python 2.7 is supported

**Comment by bjacquin** [comment]
All comments have been addressed in last PR

**Comment by doughdemon** [comment]
Thanks so far! @gentoo/proxy-maint

**Comment by gktrk** [comment]
@gentoo/proxy-maint

**Pull request closed without merging**